packages/js/src/generators/release-version/schema.json

```diff
@@ -0,0 +1,41 @@
+{
+  "$schema": "http://json-schema.org/schema",
+  "$id": "NxJSReleaseVersionGenerator",
+  "cli": "nx",
+  "title": "Implementation details of `nx release version`",
+  "description": "DO NOT INVOKE DIRECTLY WITH `nx generate`. Use `nx release version` instead.",
+  "type": "object",
+  "properties": {
+    "projects": {
+      "type": "array",
+      "description": "The ProjectGraphProjectNodes being versioned in the current execution.",
+      "items": {
+        "type": "object"
+      }
+    },
+    "projectGraph": {
+      "type": "object",
+      "description": "ProjectGraph instance"
+    },
+    "specifier": {
+      "type": "string",
+      "description": "Exact version or semver keyword to apply to the selected release group. NOTE: This should be set on the release group level, not the project level."
+    },
+    "packageRoot": {
+      "type": "string",
+      "description": "The root directory of the directory (containing a manifest file at its root) to publish. Defaults to the project root"
+    },
+    "currentVersionResolver": {
+      "type": "string",
+      "default": "disk",
+      "description": "Which approach to use to determine the current version of the project.",
+      "enum": ["registry", "disk"]
+    },
```

Comment:
What about the ability to provide a custom resolver? For instance a resolver that retrieves the version from a git tag.

Reply:
Hi @edbzn, please note the description of this PR. It is currently aimed at providing versioning and publishing for the nx repo itself only.

Rest assured we have lots of exciting plans for this coming up, which very much includes resolving versions from git tags. This is just humble beginnings and we will begin talking about more broadly once we feel it is in a good place for Nx _users_ rather than just ourselves at this point.

Reply:
Hi @JamesHenry, thank you I appreciate the clarification.

> It is currently aimed at providing versioning and publishing for the nx repo itself only.

In light of your explanation, could you please confirm whether this PR will introduce a new command that Nx users can utilize? I'm a bit confused about whether this feature will be part of the public API or not.

> This is just humble beginnings and we will begin talking about more broadly once we feel it is in a good place for Nx users rather than just ourselves at this point.

It's already looking promising, and I'm keen to provide additional feedback based on my experiences with releasing using Nx, primarily through [@jscutlery/semver](https://github.com/jscutlery/semver). However, it seems this PR may not be the ideal place for such discussions. Would you consider opening a dedicated discussion on the repository to gather input from the wider community?

Reply:
We will definitely look forward to the great feedback you will be able to provide in good time. For right now, I genuinely recommend putting this out of your mind. As noted, it is all subject to change and rapid, breaking evolution. We will begin talking about this publicly once we are ready, and at that point we will definitely seek yours and other's valuable feedback. Importantly, no existing solutions or tooling will cease to work off the back of this, no existing APIs will be negatively impacted.

Reply:
@edbzn have you had chance to join our official discord yet? https://go.nx.dev/community

I will reach out to join on there soon to discuss further